internal/pkg/agent/application/filelock/file_locker.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package filelock
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gofrs/flock"
+)
+
+var (
+	ErrZeroTimeout = errors.New("must specify a non-zero timeout for a blocking file locker")
+	ErrNotLocked   = errors.New("file not locked")
+)
+
+// FileLocker is a thin wrapper around "github.com/gofrs/flock" Flock providing both blocking and non-blocking file locking.
+// It exposes a simplified Lock*/Unlock interface and by default is non-blocking.
+// If it's not possible to acquire a lock on the specified file ErrNotLocked (directly or wrapped in another error) is returned by default.
+// It's possible to customize FileLocker behavior specifying one or more FileLockerOption at creation time
+type FileLocker struct {
+	fileLock             *flock.Flock
+	blocking             bool
+	timeout              time.Duration
+	customNotLockedError error
+}
+
+func NewFileLocker(lockFilePath string, opts ...FileLockerOption) (*FileLocker, error) {
+	flocker := &FileLocker{fileLock: flock.New(lockFilePath)}
+	for _, opt := range opts {
+		if err := opt(flocker); err != nil {
+			return nil, fmt.Errorf("applying options to new file locker: %w", err)
+		}
+	}
+	return flocker, nil
+}
+
+func (fl *FileLocker) Lock() error {
+	return fl.LockContext(context.Background())
+}
+
+func (fl *FileLocker) LockContext(ctx context.Context) error {
```

Comment:
Please add a godoc comment here explaining the default behavior w.r.t. blocking.

Reply:
see 33d8228e8ae1e255d2a9b3b35171b65f764bbda8